### ISSUE TITLE
It would be awesome to pass commands to the test engine to do matching

### ISSUE THREAD

**Issue by elsigh** [tagview/arcanist-extensions#5]
I'm using TapTestEngine to execute a command like `grunt test` which tests some js code. But I'd like to only run that test if one of the changed files matches some sort of regex. It seems like arc wants that logic to be part of the Test Engine, so maybe this could be implemented with unit.engine.tap.regex or something?

**Comment by elsigh** [comment]
Or maybe include/exclude?

**Comment by reu** [comment]
Yeah, it could be a very useful feature, and it should work for any test engine IMO.

Unfortunately I think that it won't be that easy to properly implement this, so for now, I am accepting PRs =]

**Comment by elsigh** [comment]
Curious why you don't think it will be too easy? 
I'm happy to take a crack at it and I'm just imaging code like this in run():

$regex = $this->getConfigurationManager()->getConfigFromAnySource('unit.engine.tap.regex');
$this->affectedTests = array();
foreach ($this->getPaths() as $path) {
     if (preg_match($regex, $path)) {
        // Looks like a valid test file name.
        $this->affectedTests[$path] = $path;
        continue;
      }
}

and then bail if affectedTests is empty.

but I might be oversimplifying it and not understanding something too..

(heavily cribbed from https://github.com/phacility/arcanist/blob/master/src/unit/engine/PhpunitTestEngine.php)

**Comment by elsigh** [comment]
I'll have a PR today for this.
